Neos.Flow/Classes/Reflection/ReflectionService.php

```diff
@@ -1803,32 +1803,33 @@ protected function convertParameterReflectionToArray(ParameterReflection $parame
             $parameterInformation[self::DATA_PARAMETER_ALLOWS_NULL] = true;
         }
 
-        $parameterClass = $parameter->getClass();
-        if ($parameterClass !== null) {
-            $parameterInformation[self::DATA_PARAMETER_CLASS] = $parameterClass->getName();
+        $parameterType = null;
+        if ($parameter->getType() !== null) {
+            $parameterType = $parameter->getType() instanceof \ReflectionNamedType ? $parameter->getType()->getName() : (string)$parameter->getType();
+        }
+        if ($parameter->getClass() !== null) {
+            // We use parameter type here to make class_alias usage work and return the hinted class name instead of the alias
+            $parameterInformation[self::DATA_PARAMETER_CLASS] = $parameterType;
         }
         if ($parameter->isOptional() && $parameter->isDefaultValueAvailable()) {
             $parameterInformation[self::DATA_PARAMETER_DEFAULT_VALUE] = $parameter->getDefaultValue();
         }
-        if ($method !== null) {
-            $paramAnnotations = $method->isTaggedWith('param') ? $method->getTagValues('param') : [];
-            if (isset($paramAnnotations[$parameter->getPosition()])) {
-                $explodedParameters = explode(' ', $paramAnnotations[$parameter->getPosition()]);
-                if (count($explodedParameters) >= 2) {
-                    $parameterType = $this->expandType($method->getDeclaringClass(), $explodedParameters[0]);
-                    $parameterInformation[self::DATA_PARAMETER_TYPE] = $this->cleanClassName($parameterType);
-                }
+        $paramAnnotations = $method->isTaggedWith('param') ? $method->getTagValues('param') : [];
+        if (isset($paramAnnotations[$parameter->getPosition()])) {
+            $explodedParameters = explode(' ', $paramAnnotations[$parameter->getPosition()]);
+            if (count($explodedParameters) >= 2) {
+                $parameterType = $this->expandType($method->getDeclaringClass(), $explodedParameters[0]);
```

Comment:
This is still needed to override the PHP typehint in order to allow generics/collection element typehints.